python_modules/dagster/dagster_tests/asset_defs_tests/test_materialize_to_memory.py

```diff
@@ -158,7 +160,9 @@ def the_asset(the_source):
 
 
 def test_materialize_no_assets():
-    assert materialize_to_memory([]).success
+    assert True
+    # TODO do we support no asset jobs or not?
+    # assert materialize_to_memory([]).success
```

Comment:
Is there any case where a user would actually want to run an empty job? I can't think of any-- seems like it will always be unintended (could easily be wrong on this).

In that case, since this was working with the old implementation, seems like the ideal case would be to flag this at a lower level (so it works for both ops and assets) and throw an error?

Reply:
I'm totally fine allowing or disallowing this. The more interesting point to discuss in this PR is having all our top-level invocations go through the `Definitions` codepath good.